src/bot.ts

```diff
@@ -0,0 +1,107 @@
+import { Merger } from "./github/merger";
+import { ActionLogger } from "./github/types";
+
+const BOT_COMMAND = "/bot";
+
+type Command = "merge" | "cancel" | "help";
+
+const botCommands = `
+**Available commands**
+
+- \`/bot merge\`: Enables auto-merge for Pull Request
+- \`/bot cancel\`: Cancels auto-merge for Pull Request
+- \`/bot help\`: Shows this menu
+
+For more information see the [documentation](https://github.com/paritytech/auto-merge-bot)
+`;
+
+export class Bot {
+  constructor(
+    private readonly comment: IssueComment,
+    private readonly pr: Issue,
+    private readonly logger: ActionLogger,
+    private readonly commentsApi: CommentsApi,
+  ) {}
+
+  /** Verifies if the author is the also the author of the PR or a member of the org */
+  async canTriggerBot(): Promise<boolean> {
+    this.logger.debug("Evaluating if user can trigger the bot");
+    const author = this.pr.user.id;
+    if (this.comment.user.id === author) {
+      this.logger.debug("Author of comment is also author of PR");
+      return true;
+    }
+    this.logger.debug("Author of comment is not the author of the PR");
+
+    return await this.commentsApi.userBelongsToOrg(this.comment.user.login);
+  }
+
+  async run(merger: Merger): Promise<void> {
+    this.logger.info("Running action on comment: " + this.comment.html_url);
+    if (!this.comment.body.startsWith(BOT_COMMAND)) {
+      this.logger.info(
+        `Ignoring comment ${this.comment.html_url} as it does not start with '${BOT_COMMAND}'`,
+      );
+      return;
+    }
+
+    if (this.pr.state === "closed") {
+      this.logger.info("Ignoring PR as it is closed");
+      return;
+    }
+
+    if (!(await this.canTriggerBot())) {
+      const { login } = this.comment.user;
+      const org = this.commentsApi.pullData.owner;
+      this.logger.warn(
+        "User is not allowed to trigger the bot. " +
+          `He is not the author of the PR and does not *publicly* belong to the org: https://github.com/orgs/${org}/people`,
+      );
+      await this.commentsApi.reactToComment(this.comment.id, "-1");
+      await this.commentsApi.comment(
+        "## Auto-Merge-Bot\n" +
+          `User @${login} is not the author of the PR and does not [*publicly* belong to the org \`${org}\`](https://github.com/orgs/${org}/people).\n\n` +
+          "Only author or *public* org members can trigger the bot.",
+      );
+      return;
+    }
+    this.logger.debug("User can trigger bot");
+
+    const [_, command] = this.comment.body.split(" ");
+    try {
+      switch (command as Command) {
+        case "merge":
+          await this.commentsApi.reactToComment(this.comment.id, "+1");
+          await merger.enableAutoMerge();
+          await this.commentsApi.comment(
+            "Enabled `auto-merge` in Pull Request",
+          );
+          break;
+        case "cancel":
+          await this.commentsApi.reactToComment(this.comment.id, "+1");
+          await merger.disableAutoMerge();
```

Comment:
will it be okay if it's not enabled?
or if you enable - that it's already enabled?

Reply:
Yeah, I tested that out and it doesn't change the state if it is already set. [Here you have an example](https://github.com/paritytech-stg/auto-merge-bot/pull/1#issuecomment-1735923238) where I called cancel even tho it isn't enabled.